src/cli/clean.rs

```diff
@@ -0,0 +1,144 @@
+/// Command to clean the parts of your system which are touched by pixi.
+use crate::{config, EnvironmentName, Project};
+use std::path::PathBuf;
+use std::str::FromStr;
+use std::time::Duration;
+
+use crate::progress::{global_multi_progress, long_running_progress_style};
+use clap::Parser;
+use indicatif::ProgressBar;
+
+#[derive(Parser, Debug)]
+pub enum Command {
+    #[clap(name = "cache")]
+    Cache(CacheArgs),
+}
+
+/// Clean the parts of your system which are touched by pixi.
+/// Defaults to cleaning the environments and task cache.
+/// Use the `cache` subcommand to clean the cache.
+#[derive(Parser, Debug)]
+pub struct Args {
+    #[command(subcommand)]
+    command: Option<Command>,
+    /// The path to 'pixi.toml' or 'pyproject.toml'
+    #[arg(long)]
+    pub manifest_path: Option<PathBuf>,
+
+    /// The environment directory to remove.
+    #[arg(long, short, conflicts_with = "command")]
+    pub environment: Option<String>,
+}
+
+/// Clean the cache of your system which are touched by pixi.
+#[derive(Parser, Debug)]
+pub struct CacheArgs {
+    /// Clean only the pypi related cache.
+    #[arg(long)]
+    pub pypi: bool,
+
+    /// Clean only the conda related cache.
+    #[arg(long)]
+    pub conda: bool,
+    // TODO: Would be amazing to have a --unused flag to clean only the unused cache.
+    //       By searching the inode count of the packages and removing based on that.
+    // #[arg(long)]
+    // pub unused: bool,
+}
+
+pub async fn execute(args: Args) -> miette::Result<()> {
+    match args.command {
+        Some(Command::Cache(args)) => clean_cache(args).await?,
+        None => {
+            let project = Project::load_or_else_discover(args.manifest_path.as_deref())?; // Extract the passed in environment name.
+
+            let explicit_environment = args
+                .environment
+                .map(|n| EnvironmentName::from_str(n.as_str()))
+                .transpose()?
+                .map(|n| {
+                    project.environment(&n).ok_or_else(|| {
+                        miette::miette!(
+                            "unknown environment '{n}' in {}",
+                            project
+                                .manifest_path()
+                                .to_str()
+                                .expect("expected to have a manifest_path")
+                        )
+                    })
+                })
+                .transpose()?;
+
+            if let Some(explicit_env) = explicit_environment {
+                remove_folder_with_progress(explicit_env.dir(), true).await?;
+                tracing::info!("Skipping removal of task cache and solve group environments for explicit environment '{:?}'", explicit_env.name());
```

Comment:
let's add the same tracing:info for `else` branch

Reply:
It's not really skipping anything in that part so there everything would be logged by either the progress bar or other info's